core/core.go

```diff
@@ -207,6 +217,34 @@ func (n *IpfsNode) startOnlineServices(ctx context.Context, routingOption Routin
 	return n.Bootstrap(DefaultBootstrapConfig)
 }
 
+func makeSmuxTransport(mplexExp bool) smux.Transport {
+	mstpt := mssmux.NewBlankTransport()
+
+	ymxtpt := &yamux.Transport{
+		AcceptBacklog:          8192,
+		ConnectionWriteTimeout: time.Second * 10,
+		KeepAliveInterval:      time.Second * 30,
+		EnableKeepAlive:        true,
+		MaxStreamWindowSize:    uint32(1024 * 512),
+		LogOutput:              ioutil.Discard,
+	}
+
+	mstpt.AddTransport("/yamux/1.0.0", ymxtpt)
+
+	mstpt.AddTransport("/spdy/3.1.0", spdy.Transport)
+
+	if mplexExp {
+		mstpt.AddTransport("/mplex/1.0.0", mplex.DefaultTransport)
```

Comment:
Since multiplex doesn't have a formal spec, the best we can do is follow its semver from its main implementation. See:

https://github.com/libp2p/specs/blob/master/7-properties.md#757-protocol-multicodecs

`multiplex	/multiplex/6.7.0`